### ISSUE TITLE
Packaging opentoonz in Fedora without 16 bpc TIFF support.

### ISSUE THREAD

**Issue by dherrerace** [opentoonz/opentoonz#4199]
From what I've studied in the codebase, the modification of v4.0.3 libtiff provided in the third party folder enables the library to open 16 bpc TIFF files and read 'em directly into a 16 bpc RGB buffer, enabling the aplication to use the 16 bpc data without loosing information. From what I've read, even if screens don't usually display more than 8 bpc this feature can be beneficial when image editing [1]. I don't know how impactful it can be on the normal use of this application, but I imagine that this is the purpose of this feature.

Even if this implementation is practical, it implies that opentoonz cannot benefit from the improvements of upstream libtiff without maintaining their modifications. I've read that through the years people have raised issues regarding this (#3593, #3864, #3968, #4119 #4193) and there have been different efforts to regularize this situation, either by trying to upstream the changes [2], integrating the library modifications into the codebase statically or updating the modified library.

For context, I'm working on packaging opentoonz for Fedora. The policy regarding shared libraries in Fedora is that you should use the shared libraries provided by the official repositories if they exist [3], this hasn't been a problem for me in general since most of the libraries in the third party folder are already available in Fedora, but the only exception is the modified version of libtiff. On my end, I've been thinking about some ways to handle this. This are the options that I've thought of:

1) Package opentoonz without TIFF support.

   There are no TIFF support problems if you remove TIFF support. Actually I haven't checked the impact of removing TIFF support on the codebase, but it could be an option.

2) Package opentoonz with TIFF support, but load 16 bpc TIFF files the same way as 8 bpc TIFF.

   This implies a small patch [4] and doesn't seem that invasive since the code that loads 8bpc TIFF already works loading 16bpc into an 8 bpc buffer. I tested importing 16 bpc TIFF and also opening projects that already imported 16bpc TIFFs in them and don't seem to pressent any visible problems. 

3) Just wait until a better solution is available.

   This would mean basically that I can't package this software for Fedora right now ¯\\\_(ツ)_/¯. 

What I wanted to ask is if it's ok for the community if I publish the current version of opentoonz for Fedora with patch #2 (or #1) in the mean time until this issue is resolved? 

[1] https://www.diyphotography.net/8-bit-vs-16-bit-color-depth-use-matters/

[2] https://gitlab.com/libtiff/libtiff/-/issues/264#note_592984253

[3] https://docs.fedoraproject.org/en-US/packaging-guidelines/#bundling

[4] https://pastebin.com/2trGPYRL

**Comment by RodneyBaker** [comment]
I don't think option 1 (removing Tiff support) is much of an option.
That only leaves you with option 2 but perhaps that is what you were leaning toward anyway?

**Comment by dherrerace** [comment]
Yes, it would be easy for me to publish it into the repo with that patch right away, but thought I'd ask first to see if it's not an inconvenience to publish it as it is.

I actually plan to fix the problem, but it might take more time and the app is already on a point where it can get published.

**Comment by RodneyBaker** [comment]
@dherrerace We don't have an specific policy here for closing questions that may or may not have been answered.
I will close this but feel free to re-open or start a new report as necessary.
If starting a new report it will be good to reference this one.

The community is interested in your progress so feel free to update this report even though it is closed.
We love seeing goals met and accomplished!

Thanks!

**Comment by dherrerace** [comment]
Just to update on the progress, the opentoonz-1.5 package was accepted for the next Fedora release, and if everything is ok it will be available for the current available version of Fedora by the end of the week.

Besides updating the code so it works with libtiff 4.3.0 and removing 16 bpc support as mentioned above, I had to patch a couple of other things:

* Update gethostbyname to the ipv6 compatible getaddrinfo
* Fixed installation process to use cmake variables instead of hard coded paths
* Use Fedora's provided kiss_fft static libraries instead of hard code them

Besides that, everything worked fine. The patches are available here https://src.fedoraproject.org/rpms/opentoonz/tree/rawhide

I plan to start working on upstreaming those other patches when I get an enviorment that can build opentoonz on windows (already tried to PR the gethostbyname, but failed to comply because of that...)

**Comment by dherrerace** [comment]
Just to update on this topic (It's been a while). I managed to get opentoonz 1.6.0 packaged on Fedora 37. My main delay was the addition of the tinyexr library dependency that wasn't available on Fedora, so I had to handle that too and it ended up delaying me from meeting F36 deadline.

Now I was checking what was needed to get things ready for 1.7.0, and the only problem that I'm seeing
right now is the reintroduction of [toonzrle](https://github.com/opentoonz/opentoonz/pull/4443). From what I'm seeing, registering codecs requires access to libtiff's internal API, which is not available on the packaged version of libtiff in Fedora, but since this just affects old toonz formats, I imagine it won't be a problem if I remove this from the Fedora release.
